test/libyul/yulOptimizerTests/loadResolver/keccak_reuse_basic.yul

```diff
@@ -0,0 +1,17 @@
+{
+    let x := calldataload(0)
+    let a := keccak256(0, x)
+    sstore(a, 2)
+    let t := mload(2)
+    let b := keccak256(0, x)
+    sstore(b, 3)
+}
```

Comment:
I'd add a few more tests to be able to see if the parts changed in this PR actually work correctly:
- One with `mstore` between keccak calls.
- One like the example above but with `a` ressigned to a different value after the first `sstore`.
- keccak used in an expression (e.g. `add(keccak256(a, b), keccak256(a, b))`).
- keccak used in an expression that invalidates the knowledge about keccak calls (e.g. `mstore(keccak256(a, b), keccak256(a, b))`).
- Same keccak call in all cases of a `switch`.
- Same keccak call followed by `mstore` in all cases of a `switch`.
- A call to a function containing something that invalidates memory in between same keccak calls.
- Code that contains `msize`.

Reply:
Couldn't find a good example for the switch case, but added tests for most of these scenarios.